src/src/PluginStructs/P128_data_struct.cpp

```diff
@@ -1387,36 +1388,41 @@ uint32_t P128_data_struct::rgbStr2Num(String rgbStr) {
 
 void P128_data_struct::hex2rgb(String hexcolor) {
   colorStr = hexcolor;
+  uint32_t hcolorui = rgbStr2Num(hexcolor);
+
   # if defined(RGBW) || defined(GRBW)
   hexcolor.length() <= 6
-    ? rgb = RgbColor(rgbStr2Num(hexcolor) >> 16, rgbStr2Num(hexcolor) >> 8, rgbStr2Num(hexcolor))
-    : rgb = RgbwColor(rgbStr2Num(hexcolor) >> 24, rgbStr2Num(hexcolor) >> 16, rgbStr2Num(hexcolor) >> 8, rgbStr2Num(hexcolor));
+    ? rgb = RgbColor(hcolorui >> 16, hcolorui >> 8, hcolorui)
+    : rgb = RgbwColor(hcolorui >> 24, hcolorui >> 16, hcolorui >> 8, hcolorui);
   # else // if defined(RGBW) || defined(GRBW)
-  rgb = RgbColor(rgbStr2Num(hexcolor) >> 16, rgbStr2Num(hexcolor) >> 8, rgbStr2Num(hexcolor));
+  rgb = RgbColor(hcolorui >> 16, hcolorui >> 8, hcolorui);
   # endif // if defined(RGBW) || defined(GRBW)
 }
 
 void P128_data_struct::hex2rrggbb(String hexcolor) {
```

Comment:
`const String& hexcolor` as function parameter.
Now it is copied when calling the function.